controller/persistence/transit_router_store.go

```diff
@@ -0,0 +1,204 @@
+	"go.etcd.io/bbolt"
+	"reflect"
+)
+
+const (
+	FieldTransitRouterIsVerified  = "isVerified"
+	FieldTransitRouterEnrollments = "enrollments"
+
+	//old
+	FieldTransitRouterEnrollmentToken     = "enrollmentToken"
+	FieldTransitRouterEnrollmentJwt       = "enrollmentJwt"
+	FieldTransitRouterEnrollmentCreatedAt = "enrollmentCreatedAt"
+	FieldTransitRouterEnrollmentExpiresAt = "enrollmentExpiresAt"
+)
+
+type TransitRouter struct {
+	db.Router
+	Name        string
+	IsVerified  bool
+	Enrollments []string
+	IsBase      bool
+}
+
+func (entity *TransitRouter) GetId() string {
+	return entity.Id
+}
+
+func (entity *TransitRouter) SetId(id string) {
+	entity.Id = id
+}
+
+func newTransitRouter(name string, roleAttributes ...string) *TransitRouter {
+	return &TransitRouter{
+		Router: db.Router{
+			BaseExtEntity: boltz.BaseExtEntity{Id: uuid.New().String()},
+		},
+		Name: name,
+	}
+}
+
+func (entity *TransitRouter) LoadValues(store boltz.CrudStore, bucket *boltz.TypedBucket) {
+	_, err := store.GetParentStore().BaseLoadOneById(bucket.Tx(), entity.Id, &entity.Router)
+	bucket.SetError(err)
+
+	if bucket.Bucket == nil {
+		entity.IsVerified = true
+		entity.IsBase = true
+		return
+	}
+	entity.LoadBaseValues(bucket)
+	entity.Name = bucket.GetStringOrError(FieldName)
+	entity.IsVerified = bucket.GetBoolWithDefault(FieldTransitRouterIsVerified, false)
+	entity.Enrollments = bucket.GetStringList(FieldTransitRouterEnrollments)
+}
+
+func (entity *TransitRouter) SetValues(ctx *boltz.PersistContext) {
+	entity.Router.SetValues(ctx.GetParentContext())
+	entity.SetBaseValues(ctx)
+	ctx.SetString(FieldName, entity.Name)
+}
+
+func (entity *TransitRouter) GetEntityType() string {
+	return EntityTypeTransitRouters
+}
+
+func (entity *TransitRouter) GetName() string {
+	return entity.Name
+}
+
+type TransitRouterStore interface {
+	NameIndexedStore
+	LoadOneById(tx *bbolt.Tx, id string) (*TransitRouter, error)
+	LoadOneByName(tx *bbolt.Tx, id string) (*TransitRouter, error)
+}
+
+func newTransitRouterStore(stores *stores) *transitRouterStoreImpl {
+	store := &transitRouterStoreImpl{
+		baseStore: newExtendedBaseStore(stores, stores.Router, EntityTypeTransitRouters),
+	}
+	store.InitImpl(store)
+	return store
+}
+
+type transitRouterStoreImpl struct {
+	*baseStore
+	indexName         boltz.ReadIndex
+	symbolEnrollments boltz.EntitySetSymbol
+}
+
+func (store *transitRouterStoreImpl) NewStoreEntity() boltz.Entity {
+	return &TransitRouter{}
+}
+
+func (store *transitRouterStoreImpl) initializeLocal() {
+	store.AddExtEntitySymbols()
+	store.GetParentStore().GrantSymbols(store)
+
+	store.indexName = store.addUniqueNameField()
+	store.symbolEnrollments = store.AddFkSetSymbol(FieldTransitRouterEnrollments, store.stores.enrollment)
+}
+
+func (store *transitRouterStoreImpl) initializeLinked() {
+
+	store.AddLinkCollection(store.symbolEnrollments, store.stores.enrollment.symbolIdentity)
+
+	//todo: confirm this is needed
```

Comment:
I don't think this is needed. Should delete the base store bucket which will clear everything underneath.

Reply:
Fixed